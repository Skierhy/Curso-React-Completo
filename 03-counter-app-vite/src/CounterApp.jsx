import { useState } from 'react';
import PropTypes from 'Prop-types';

export const CounterApp = ({ value }) => {
	const [count, setCount] = useState(value);

	const counter = () => {
		setCount(count + 1);
	};

	const counterMenos = () => {
		setCount(count - 1);
	};

	const reset = () => {
		setCount(value);
	};

	return (
		<>
			<h1>CounterApp </h1>
			<h2> {count} </h2>
			<button onClick={counter}>Contador +1</button>
			<button onClick={counterMenos}>Contador -1</button>
			<button aria-label="btn-reset" onClick={reset}>
				Reset
			</button>
		</>
	);
};

CounterApp.propTypes = {
	value: PropTypes.number.isRequired,
};

CounterApp.defaultProps = {
	value: '4444',
};
