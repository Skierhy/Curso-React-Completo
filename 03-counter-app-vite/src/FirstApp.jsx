import { PropTypes } from 'prop-types';

// const nombreApp = 'Hola SkierhyXD';
// const obj = {
// 	nombre: 'SkierhyXD',
// 	edad: 23,
// };

// const nombre = () => {
// 	return 'Soy una función flecha';
// };
export const FirstApp = ({ title, subTitle, name }) => {
	return (
		<>
			{/* <h1>Hola {title}</h1>
			<p>Mi primera aplicación</p>
			<h2>{nombreApp}</h2>
			<h3>{nombre()}</h3>
			<code>{JSON.stringify(obj)}</code>
			<br />
			<h3>{title}</h3>
			<h3>{numero}</h3> */}
			<h1 data-testid="test-title">{title}</h1>
			<p>{subTitle}</p>
			<p>{subTitle}</p>
			<p>{name}</p>
		</>
	);
};

FirstApp.propTypes = {
	title: PropTypes.string.isRequired,
	subTitle: PropTypes.string,
};

FirstApp.defaultProps = {
	name: 'Skierhy',
	subTitle: 'No hay subtítulo',
	// title: 'No hay título',
};
