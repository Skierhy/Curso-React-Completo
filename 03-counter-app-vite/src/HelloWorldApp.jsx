// // funcional component
// // actualmente no se usan las clases
// function App() {
// 	// document.createElement('h1')
// 	// .innerHTML = 'Hello World'
// 	return <h1>hola mundo</h1>;
// }

// export default App;

export const HelloWorldApp = () => {
	return <h1>Hello World App</h1>;
};
