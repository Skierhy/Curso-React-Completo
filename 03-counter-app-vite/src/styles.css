html,
body {
	background-color: #21232a;
	color: #fff;
	font-family: Arial, Helvetica, sans-serif;
	font-size: 1.3rem;
	padding: 7rem;
}

button {
	padding: 1.5rem;
	font-size: 1.3rem;
	margin-right: 0.5rem;
}
