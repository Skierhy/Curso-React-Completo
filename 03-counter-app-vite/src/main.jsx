// react es una librería de javascript
import React from 'react';
// para renderizar la aplicación
import ReactDOM from 'react-dom/client';
// importar el componente App
import { CounterApp } from './CounterApp';
// import { FirstApp } from './FirstApp';
// import { HelloWorldApp } from './HelloWorldApp';

// importar el css global
import './styles.css';

// renderizar la aplicación
ReactDOM.createRoot(document.getElementById('root')).render(
	<React.StrictMode>
		{/* <HelloWorldApp /> */}
		{/* <FirstApp title="Hello World" /> */}
		<CounterApp value={123} />
	</React.StrictMode>
);
