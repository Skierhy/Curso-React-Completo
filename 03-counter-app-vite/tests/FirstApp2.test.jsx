import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Prueba en FirstApp', () => {
	const title = 'Hola soy el título';
	const subTitle = 'Hola soy el subtítulo';

	test('debe que hacer match con el snapshot', () => {
		const { container } = render(<FirstApp title={title} />);
		expect(container).toMatchSnapshot();
	});

	test('debe de mostrar el mensaje "Hola soy el título"', () => {
		render(<FirstApp title={title} />);
		expect(screen.getByText(title)).toBeTruthy();
		// expect(screen.getByText(title)).not.toBeTruthy();
	});

	test('debe de mostrar el titulo en un h1', () => {
		render(<FirstApp title={title} />);
		// getByRole es una forma de buscar elementos en el DOM
		expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(
			title
		);
	});

	test('debe de mostrar el subtítulo enviado por props', () => {
		render(<FirstApp title={title} subTitle={subTitle} />);
		expect(screen.getAllByText(subTitle).length).toBe(2);
	});
});
