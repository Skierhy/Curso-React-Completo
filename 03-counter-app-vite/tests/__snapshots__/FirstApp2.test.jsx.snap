// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en FirstApp debe que hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy el título
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Skierhy
  </p>
</div>
`;
